pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -469,6 +470,83 @@ public LocalDateTime toLocalDateTime(String s) throws SQLException {
       return result;
     }
   }
+
+  /**
+   * Parse a string and return a LocalDateTime representing its value.
+   *
+   * @param s The ISO formated date string to parse.
+   * @return null if s is null or a LocalDateTime of the parsed string s.
+   * @throws SQLException if there is a problem parsing s.
+   */
+  public OffsetDateTime toOffsetDateTime(String s) throws SQLException {
+    if (s == null) {
+      return null;
+    }
+
+    int slen = s.length();
+
+    // convert postgres's infinity values to internal infinity magic value
+    if (slen == 8 && s.equals("infinity")) {
+      return OffsetDateTime.MAX;
```

Comment:
are these the same numbers ? In other words if I read infiinity can I write it back out again

Reply:
That is a good point. I added test for this, the code for `-infinity` was not based around the minimum value for Postgres.